shared/dataflow/codeql/dataflow/internal/FlowSummaryImpl.qll

```diff
@@ -665,22 +678,33 @@ module Make<
     pragma[nomagic]
     private predicate derivedFluentFlow(
       SummarizedCallable c, SummaryComponentStack input, SummaryComponentStack arg,
-      SummaryComponentStack return, boolean preservesValue
+      SummaryComponentStack return, boolean preservesValue, string model
     ) {
-      exists(ParameterPosition pos |
-        summary(c, input, arg, preservesValue) and
+      exists(ParameterPosition pos, string model1, string model2 |
+        summary(c, input, arg, preservesValue, model1) and
         isContentOfArgument(arg, pos) and
-        summary(c, SummaryComponentStack::argument(pos), return, true) and
-        return.bottom() = TReturnSummaryComponent(_)
+        summary(c, SummaryComponentStack::argument(pos), return, true, model2) and
+        return.bottom() = TReturnSummaryComponent(_) and
+        model = mergeModels(model1, model2)
       )
     }
 
+    bindingset[model1, model2]
+    pragma[inline_late]
+    private string mergeModels(string model1, string model2) {
```

Comment:
This actually occurs with a decent frequency and is likely to be limited to two models - and I generally want to see the two contributing MaDids for derived models. The merging inside data flow is different because it's both unlikely to occur, and at the same time not really bounded in any way. Ideally we'd use a set, but that's not really possible (at least it would be extremely messy to try to implement a set data structure in a string).